### PR TITLE
Simplify lookup, browse and search functions

### LINKED ISSUE #822: BUG Property 'recordings' does not exist on type 'ISearchResult'.
```
const rawPageTracks = await api.search('recording',{query:MYQUERY});
console.log(rawPageTracks.recordings);
```

gives me 
 Property 'recordings' does not exist on type 'ISearchResult'.

casting it to `any` works.

```
const rawPageTracks = await api.search('recording',{query:MYQUERY}) as any;
```

### PR DESCRIPTION
**Pull request by Borewit**
Remove specific search function in favor of a single typed function: 

- `mbApi.browseArtist(query)`  with `mbApi.browse('artist', query)` 
- `mbApi.lookupArtist('ab2528d9-719f-4261-8098-21849222a0f2')` with `mbApi.lookup('artist', 'ab2528d9-719f-4261-8098-21849222a0f2')` 
- `mbApi.searchArtist({query: 'Stromae'})`  with `mbApi.search('artist', {query: 'Stromae'})` 

Resolves #822

### CODE REVIEW

**Comment by Borewit** [comment]
If you are happy with this one @Haschikeks , I like to merge it.

File: lib/musicbrainz-api.ts [hunk 1]
```diff
@@ -266,256 +266,55 @@ export class MusicBrainzApi {
...
-   * @param mbid
-   * @param inc
-   */
-  public lookupEntity<T, I extends string = never>(entity: mb.EntityType, mbid: string, inc: I[] = []): Promise<T> {
+   * Lookup entity
+   * @param entity 'area', 'artist', collection', 'instrument', 'label', 'place', 'release', 'release-group', 'recording', 'series', 'work', 'url' or 'event'
+   * @param mbid Entity MBID
+   * @param inc Query, like: {<entity>: <MBID:}
+   */
+  public lookup<I extends string = never>(entity: 'area', areaId: string, inc?: I[]): Promise<mb.IArea>;
```
[the remark this answers is not in the document]
Comment (Borewit) [reply lib/musicbrainz-api.ts]: Thanks for pointing that out. No indeed, the `inc` should be typed in line with the entity type.

**Review by Haschikeks — APPROVED** [review]
Looks good.
But we should keep in mind, that this is a breaking API change.

**Comment by Borewit** [comment]
> But we should keep in mind, that this is a breaking API change.

Very true.
I will write add a warning in the release notes. 
We are still releasing as 0.x.x versions, which means the API is subject to change.
Although I think this one is received well, we should soon release 1.0.0.

Thanks for your review @Haschikeks, much appreciated.

**Pull request merged**